tests/unit/test_utils.py

```diff
@@ -1,11 +1,93 @@
+import datetime
+
 import numpy as np
+import pandas as pd
 import pytest
 import xarray as xr
 from packaging.version import Version
 
 import mesmer.core.utils
 
 
+def make_dummy_yearly_data(freq):
+    if freq == "YM":
+        time = [datetime.datetime(i, 7, 15) for i in range(2000, 2005)]
+    else:
+        time = xr.cftime_range(start="2000", periods=5, freq=freq)
+
+    data = xr.DataArray([1, 2, 3, 4, 5], dims=("time"), coords={"time": time})
+    return data
+
+
+def make_dummy_monthly_data(freq):
+    if freq == "MM":
+        time = [
+            datetime.datetime(i, j, 15) for i in range(2000, 2005) for j in range(1, 13)
+        ]
+    else:
+        time = xr.cftime_range(start="2000-01", periods=5 * 12, freq=freq)
+
+    data = xr.DataArray(np.ones(5 * 12), dims=("time"), coords={"time": time})
+    return data
+
+
+@pytest.mark.parametrize("freq_y", ["YM", "YS", "YE", "YS-JUL", "YS-NOV"])
+@pytest.mark.parametrize("freq_m", ["MM", "MS", "ME"])
+def test_upsample_yearly_data(freq_y, freq_m):
+    if Version(pd.__version__) < Version("2.2"):
+        if freq_y == "YE":
+            freq_y = freq_y.replace("YE", "A")
+        elif "YS" in freq_y:
+            freq_y = freq_y.replace("YS", "AS")
+
+        if freq_m == "ME":
+            freq_m = freq_m.replace("ME", "M")
+
+    yearly_data = make_dummy_yearly_data(freq_y)
+    monthly_data = make_dummy_monthly_data(freq_m)
+
+    upsampled_years = mesmer.core.utils.upsample_yearly_data(
+        yearly_data, monthly_data.time
+    )
+
+    assert (upsampled_years.time == monthly_data.time).all()
```

Comment:
```suggestion
    xr.testing.assert_equal(upsampled_years.time, monthly_data.time)
```

I think that should work